dds/DCPS/transport/framework/TransportQueueElement.h

```diff
@@ -160,6 +160,13 @@ class OpenDDS_Dcps_Export TransportQueueElement : public PoolAllocationBase {
 
   virtual bool is_retained_replaced() const { return false; }
 
+  struct OrderBySequenceNumber {
+    bool operator()(const TransportQueueElement* lhs, const TransportQueueElement* rhs) const
+    {
+      return lhs->sequence() < rhs->sequence();
```

Comment:
Add a fallback to pointer comparison when sequence numbers are equal (assuming this is possible?)
```suggestion
      return lhs->sequence() < rhs->sequence() || (lhs->sequence() == rhs->sequence() && lhs < rhs);
```

Reply:
I don't believe that is possible.  `to_deliver` is populated from `elems_not_acked_` which is a map from sequence number to TQE.  Thus, it looks like we are already assuming a one-to-one correspondence with sequence number.  If anything, we should add asserts when we insert into `elems_not_acked_` if we believe this is a problem.  Other transports may be different but they aren't using this code yet.

However, I'm okay with the extra check.